lib/preprocess.js

```diff
@@ -89,15 +92,37 @@ export const preprocess = async ({
   })
 }
 
-export const fetchMeasurementsViaClient = async (web3Storage, cid) => {
-  const res = await web3Storage.get(cid)
-  const files = await res.files()
-  const measurements = JSON.parse(await files[0].text())
-  return measurements
+export const fetchMeasurementsTrustless = async cid => {
+  const res = await fetch(
+    `https://${encodeURIComponent(cid)}.ipfs.w3s.link?format=car`
+  )
+  if (!res.ok) {
+    const msg = `Cannot fetch measurements ${cid}: ${res.status}\n${await res.text()}`
+    throw new Error(msg)
+  }
+  const reader = await CarReader.fromIterable(res.body)
+  const entries = exporter(cid, {
+    async get (cid) {
+      const block = await reader.get(cid)
+      await validateBlock(block)
+      return block.bytes
+    }
+  })
+  for await (const entry of entries) {
+    if (entry.type === 'file' || entry.type === 'raw') {
```

Comment:
> Files are now published to the root entry of the CID (vs at /measurements.json), therefore this PR and the spark-api PR need to be merged together.

What are the implication of this for the upgrade path and for the evaluation of the round when we deploy these changes? Is it a lot of work to add support for `/measurements.json`?

Reply:
When we deploy a new `spark-evaluate` instance we sacrifice that round anyway, due to the in-memory storage. I think therefore this is fine. In the future once we use sqlite we need to think about this more